runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -189,8 +189,8 @@ J9::Z::TreeEvaluator::inlineStringLatin1Inflate(TR::Node *node, TR::CodeGenerato
 
    // Done storing 16 chars. Now do some bookkeeping and then branch back to start label.
    generateRILInstruction(cg, TR::InstOpCode::AFI, node, srcOffRegister, 16);
-   generateRILInstruction(cg, TR::InstOpCode::AFI, node, sourceArrayReferenceRegister, 16);
-   generateRILInstruction(cg, TR::InstOpCode::AFI, node, charArrayReferenceRegister, 32);
+   generateRILInstruction(cg, cg->comp()->target().is64Bit() ? TR::InstOpCode::AGFI : TR::InstOpCode::AFI, node, sourceArrayReferenceRegister, 16);
+   generateRILInstruction(cg, cg->comp()->target().is64Bit() ? TR::InstOpCode::AGFI : TR::InstOpCode::AFI, node, charArrayReferenceRegister, 32);
```

Comment:
Arguably, we could be using `LA sourceArrayReferenceRegister, 16/32(sourceArrayReferenceRegister)` here instead to compute the updated address.   That will handle both 31-bit and 64-bit patterns, and LA has a nicer bypass in the hardware when the target register is fed into memory references when consumed by the Load-Store Unit.

Reply:
Using LA now in 24cd8901ab3f18e45e45fb3cc5df0c6843b045b2